R/03b-plot-catchments.R

```diff
@@ -0,0 +1,499 @@
+  mutate(across(c(csd_name), as.character),
+         across(c(landarea), as.numeric))
+
+# Get centroids for labels
+csd_centroids <- st_centroid(csd_shapefile) 
+csd_centroids_nudged <- csd_centroids |> 
+  mutate(
+      geom = case_when(
+        csd_name == 'Dawson Creek' ~ geom + c(-150000, 40000),
+        csd_name == 'Langford' ~ geom + c(130000, 10000), # move east for langford
+        TRUE ~ geom + c(0, 70000)  # move label north to not overlap 
+      )
+  ) %>%
+  st_set_crs(st_crs(csd_centroids))
+
+# Locations of all Service BC locations 
+sbc_locs <- read_csv(glue("{SRC_DATA_FOLDER}/full-service-bc-locs.csv")) |>
+  st_as_sf(
+    coords = c('coord_x', 'coord_y'),
+    crs = 3005
+  ) 
+
+# DB shapefiles
+db_shapefile <-
+  st_read(glue("{SHAPEFILE_OUT}/full-db_with_location.gpkg")) %>%
+  mutate(across(c(landarea), as.numeric))
+
+# Read the pre-computed assignment data 
+complete_assignments <- read_csv(glue("{SRC_DATA_FOLDER}/complete_db_assignments.csv"))
+
+# Filter to relevant datasets for the maps
+csds <- csd_shapefile |>
+  filter(csd_name %in% CSD_NAMES)
+
+csd_centroids_nudged <- csd_centroids_nudged |>
+  filter(csd_name %in% CSD_NAMES)
+
+pilot_sbc_locs <- sbc_locs |>
+  filter(csd_name %in% CSD_NAMES)
+
+  # Save the maps to the outputs directory
+maps_folder <- file.path(MAP_OUT, "complete_catchments")
+if (!dir.exists(maps_folder)) {
+  dir.create(maps_folder, recursive = TRUE)
+}
+
+# ----------------------------------------------------------------------------
+# Map 1: Pilot regions overview
+# ----------------------------------------------------------------------------
+
+pilot_map <- ggplot() + 
+  geom_sf(
+    data = bc_map, 
+    fill = 'white', 
+    color = 'black'
+  ) +
+  geom_sf(
+    data = csds, 
+    aes(fill = as.factor(csd_name)),
+    color = 'darkgrey',
+    alpha = 1
+  ) +
+  geom_sf(
+    data = pilot_sbc_locs,
+    shape = 23,
+    fill = 'yellow',
+    color = 'black',
+    size = 2,
+    stroke = 1.1
+  ) +
+  geom_sf_text(
+    data = csd_centroids_nudged,
+    aes(label = csd_name), 
+    size = 4, 
+    fontface = 'bold'
+  ) +
+  scale_fill_brewer(
+    palette = 'Set2',
+    name = 'Locality'
+  ) +
+  coord_sf(expand = FALSE) + # remove whitespace
+  labs(
+    title = 'Pilot Regions'
+  ) +
+  MAP_THEME +
+  theme(
+    legend.position = "none",
+    axis.title = element_blank()
+  )
+
+pilot_map
+
+# Save the pilot regions map
+ggsave(
+  filename = glue("pilot_regions.png"),
+  path = maps_folder,
+  plot = pilot_map,
+  width = 8,
+  height = 7,
+  dpi = 300
+)
+
+# ----------------------------------------------------------------------------
+# Map 2: Original drive-time based catchments (with unassigned areas)
+# ----------------------------------------------------------------------------
+
+# Create original catchment areas (with unassigned areas)
+original_catchments <- db_shapefile |>
+  left_join(complete_assignments, by = "dbid") |>
```

Comment:
I'm receiving this error after attempting to join: 

`! Can't join `x$dbid` with `y$dbid` due to incompatible types.
ℹ `x$dbid` is a <character>.
ℹ `y$dbid` is a <double>.`